test/UnitTestFramework/UnitTestFramework.js

```diff
@@ -167,8 +191,120 @@ var testRunner = function testRunner() {
+            asyncTest.error = new Array(testsToRun.length).fill("");
+
+            if (options) {
+                if (options.hasOwnProperty("verbose")) {
+                    _verbose = options.verbose;
+                }
+            }
+
+            for (var i in testsToRun) {
+                var isRunnable = typeof testsToRun[i] === objectType;
+                if (isRunnable) {
+                    this.runAsyncTest(i, testsToRun[i].name, testsToRun[i].body);
+                }
+            }
+
+            if (_verbose || executedTestCount != passedTestCount) {
+                helpers.writeln("Summary of async tests: total load attempts: ", executedTestCount,
+                  "; loaded: ", passedTestCount, "; failed: ", executedTestCount - passedTestCount);
+            }
+
+            Promise.all(asyncTest.promise.map(Promise.all, Promise)).then( function() {
+                passedTestCount = 0;
+                for (var i = 0; i < asyncTest.count.length; i++) {
+                    var passed = testRunner.asyncTestHasNoErr(i);
+                    if (_verbose) {
+                        helpers.writeln("*** Result of test #", i + 1, " (", i, "): ", testsToRun[i].name);
+                        if (!passed) helpers.writeln(asyncTest.error[i]);
+                        helpers.writeln(passed ? "PASSED" : "FAILED");
+                    }
+                    !passed || passedTestCount++;
+                }
+                if (_verbose || executedTestCount != passedTestCount) {
+                    helpers.writeln("Summary of tests: total executed: ", executedTestCount,
+                      "; passed: ", passedTestCount, "; failed: ", executedTestCount - passedTestCount);
+                } else {
+                    helpers.writeln("pass");
+                }
+            });
+        },
+
+        runAsyncTest: function runAsyncTest(testIndex, testName, testBody) {
+            ///<summary>Runs test body catching all exceptions.<br/>
+            /// Result: prints LOADED/LOADING FAILED to the output.
+            ///</summary>
+            function logTestNameIf(b) {
+                if (b) {
+                    helpers.writeln("*** Loading test #", executedTestCount + 1, " (", testIndex, "): ", testName);
+                }
+            }
+
+            logTestNameIf(_verbose);
+
+            asyncTest.testIndex = testIndex;
+            asyncTest.promise[testIndex] = [];
+            asyncTest.resolve[testIndex] = [];
+
+            var isSuccess = true;
+            try {
+                testBody(testIndex);
+            } catch (ex) {
+                var message = ex.stack || ex.message || ex;
+                logTestNameIf(!_verbose);
+                helpers.writeln("Test threw exception: ", message);
+                isSuccess = false;
+            }
+            if (isSuccess) {
+                if (_verbose) {
+                    helpers.writeln("LOADED");
+                }
+                ++passedTestCount;

```

Comment:
In the new rev: merging runAsyncTests and runAsyncTest back into runTests and runTest.


---
In reply to: [110792175](https://github.com/Microsoft/ChakraCore/pull/2822#discussion_r110792175) [](ancestors = 110792175,110760516)